ambry-router/src/test/java/com/github/ambry/router/CompressionServiceTest.java

```diff
@@ -0,0 +1,341 @@
+    Triple<byte[], Integer, Integer> bufferInfo = service.convertByteBufToByteArray(sourceBuffer);
+    Assert.assertEquals(arrayBuffer, bufferInfo.getLeft());
+    Assert.assertEquals(3, (int) bufferInfo.getMiddle());
+    Assert.assertEquals(8, (int) bufferInfo.getRight());
+
+    // Test: Convert wrapped ByteBuffer buffer to byte[]
+    sourceBuffer = Unpooled.wrappedBuffer(ByteBuffer.wrap(arrayBuffer, 3, 8));
+    bufferInfo = service.convertByteBufToByteArray(sourceBuffer);
+    Assert.assertEquals(arrayBuffer, bufferInfo.getLeft());
+    Assert.assertEquals(3, (int) bufferInfo.getMiddle());
+    Assert.assertEquals(8, (int) bufferInfo.getRight());
+
+    // Test: Convert multiple wrapped buffers to byte[].
+    byte[] arrayBuffer1 = new byte[] { -2, -1, 0, };
+    byte[] arrayBuffer2 = new byte[] { 1, 2, 3, 4, 5, 6, 7, 8, 9 };
+    sourceBuffer = Unpooled.wrappedBuffer(arrayBuffer1, arrayBuffer2);
+    bufferInfo = service.convertByteBufToByteArray(sourceBuffer);
+    Assert.assertNotEquals(arrayBuffer, bufferInfo.getLeft());
+    Assert.assertEquals(0, (int) bufferInfo.getMiddle());
+    Assert.assertEquals(12, (int) bufferInfo.getRight());
+  }
+
+  @Test
+  public void testCompressThenDecompress() throws CompressionException {
+    byte[] sourceBuffer = ("Test Message for testing purpose.  The Message is part of the testing message."
+        + "Test Message for testing purpose.  The Message is part of the testing message."
+        + "Test Message for testing purpose.  The Message is part of the testing message.").getBytes();
+
+    // Decompressed full chunk.
+    CompressionConfig config = new CompressionConfig();
+    config.minimalSourceDataSizeInBytes = 1;
+    config.minimalCompressRatio = 1.0;
+    CompressionMetrics metrics = new CompressionMetrics(new MetricRegistry());
+    CompressionService service = new CompressionService(config, metrics);
+
+    // Compress the source buffer.
+    ByteBuf compressedBuffer = service.compressChunk(Unpooled.wrappedBuffer(sourceBuffer), false);
+    Assert.assertNotEquals(sourceBuffer.length, compressedBuffer.readableBytes());
+    Assert.assertEquals(1, metrics.compressAcceptRate.getCount());
+
+    // Decompress the source buffer.
+    ByteBuf decompressedBuffer = service.decompress(compressedBuffer, sourceBuffer.length);
+    Assert.assertEquals(sourceBuffer.length, decompressedBuffer.readableBytes());
+    Assert.assertEquals(1, metrics.decompressSuccessRate.getCount());
+    Assert.assertTrue(metrics.decompressExpandSizeBytes.getCount() > 0);
+    Assert.assertArrayEquals(decompressedBuffer.array(), sourceBuffer);
+
+    // Zstd is the default compression algorithm.
+    CompressionMetrics.AlgorithmMetrics algorithmMetrics = metrics.getAlgorithmMetrics(ZstdCompression.ALGORITHM_NAME);
+    Assert.assertEquals(1, algorithmMetrics.decompressRate.getCount());
+    Assert.assertTrue(algorithmMetrics.fullSizeDecompressTimeInMicroseconds.getCount() > 0);
+  }
+}
```

Comment:
Test with unknown algorithm should default to zstd

Reply:
Line 148 tests invalid algorithm name and it fell back to Zstd.  There is already a test case for it.  This case is when the compression algorithm is not specified in config, it would use the default Zstd.